cmd/snap-seccomp/main_test.go

```diff
@@ -217,26 +232,39 @@ set_tls
 # arm64
 readlinkat
 faccessat
+# i386 from amd64
+restart_syscall
 `
 	bpfPath := filepath.Join(c.MkDir(), "bpf")
 	err = main.Compile([]byte(common+seccompWhitelist), bpfPath)
 	c.Assert(err, IsNil)
 
+	// default syscall runner
+	syscallRunner := s.seccompSyscallRunner
+
 	// syscallName;arch;arg1,arg2...
 	l := strings.Split(bpfInput, ";")
+	syscallNr, err := seccomp.GetSyscallFromName(l[0])
+	c.Assert(err, IsNil)
 	if len(l) > 1 && l[1] != "native" {
-		c.Logf("cannot use non-native in runBpfInKernel")
-		return
+		syscallNr, err = seccomp.GetSyscallFromNameByArch(l[0], main.UbuntuArchToScmpArch(l[1]))
+		c.Assert(err, IsNil)
+		// switch to the 32bit version of the syscall runner
+		syscallRunner = s.seccompSyscallRunner + ".m32"
```

Comment:
I don't understand the details of what is going on here, so apologies if this is completely off: can we really imply that any time the syscall is not native anywhere, having a 32bit runner is the right thing to do? I see no check above verifying anything about the actual architectures involved.

Reply:
Indeed, we control the test inputs so this works but its very non-obvious, I pushed a new revision that makes this more explicit and also adds a comment with the details about the architecture handling.